### PR TITLE
lz4.go: Fix Go 1.16 SliceHeader vet warnings

### PR DESCRIPTION
**Pull request by evanj**
Go 1.16 added vet warnings to check for unsafe uses of
reflect.SliceHeader. To fix them, create an empty slice, then alter
that slice using *reflect.SliceHeader. This is apparently safer
because it works with escape analysis and garbage collection [1].
This pattern is also used in golang.org/x/sys so this should be safe.
For example, see Mmap in syscall_unix.go:

https://cs.opensource.google/go/x/sys/+/master:unix/syscall_unix.go

[1] https://github.com/golang/go/issues/40701

Fixes:

./lz4.go:162:35: possible misuse of reflect.SliceHeader
./lz4.go:509:35: possible misuse of reflect.SliceHeader

### CODE REVIEW

File: lz4.go [hunk 1]
```diff
@@ -154,12 +154,12 @@ func (w *Writer) writeFrame(src []byte) (int, error) {
...
-		Len:  streamingBlockSize,
-		Cap:  streamingBlockSize,
-	}
-	return *(*[]byte)(unsafe.Pointer(&tmpSlice))
+	var result []byte
+	resultHeader := (*reflect.SliceHeader)(unsafe.Pointer(&result))
+	resultHeader.Data = uintptr(w.compressionBuffer[w.inpBufIndex])
+	resultHeader.Len = streamingBlockSize
+	resultHeader.Cap = streamingBlockSize
+	return result
```
Comment (HippoBaro) [inline lz4.go]: We should use [`unsafe.Slice`](https://pkg.go.dev/unsafe#Slice) now that it's available to us

File: lz4.go [hunk 2]
```diff
@@ -501,10 +501,10 @@ func (r *DecompressReader) readSize(rdr io.Reader) (int, error) {
...
-		Len:  _len,
-		Cap:  _cap,
-	}
-	return *(*[]byte)(unsafe.Pointer(&tmpSlice))
+	var result []byte
+	resultHeader := (*reflect.SliceHeader)(unsafe.Pointer(&result))
+	resultHeader.Data = uintptr(dataPtr)
+	resultHeader.Len = _len
+	resultHeader.Cap = _cap
+	return result
```
Comment (HippoBaro) [inline lz4.go]: ditto

**Review by HippoBaro — CHANGES_REQUESTED** [review]

File: lz4.go (hunk 1 above)
Comment (evanj) [reply lz4.go]: I totally forgot about this, thanks! This is much better. This now requires Go 1.17, but I think that is okay also. Fixed! I'll change the commit comment to match also.

**Review by HippoBaro — APPROVED** [review]
Awesome!

**Pull request merged**